inference-engine/thirdparty/clDNN/kernel_selector/core/kernel_selector.cpp

```diff
@@ -242,25 +243,36 @@ KernelsData kernel_selector_base::GetAutoTuneBestKernel(const Params& params,
 }
 
 KernelList kernel_selector_base::GetAllImplementations(const Params& params, const optional_params& options, KernelType kType) const {
+    using PriorityPair = std::pair<KernelsPriority, std::shared_ptr<KernelBase>>;
+    std::vector<PriorityPair> sortedImpls;
     KernelList result;
 
     if (params.GetType() == kType && options.GetType() == kType) {
         ParamsKey requireKey = params.GetParamsKey().Merge(options.GetSupportedKey());
         bool forceImplementation = !params.forceImplementation.empty();
-
-        std::copy_if(
-            implementations.begin(),
-            implementations.end(),
-            std::back_inserter(result),
-            [&](const std::shared_ptr<KernelBase>& implementation) {
-            const ParamsKey implKey = implementation->GetSupportedKey();
+        for (auto& impl : implementations) {
+            const ParamsKey implKey = impl->GetSupportedKey();
             if (!implKey.Support(requireKey))
-                return false;
-            if (forceImplementation && params.forceImplementation != implementation->GetName())
-                return false;
+                continue;
+            if (forceImplementation && params.forceImplementation != impl->GetName())
+                continue;
+            sortedImpls.emplace_back(impl->GetKernelsPriority(params, options), impl);
+        }
 
-            return true;
-        });
+        stable_sort(
```

Comment:
If I remember correctly, we agreed to insert implementation into `sortedImpls` in sorted order and get rid of separate sort pass. I think std::set with custom comparator for `PriorityPair` will do the trick.

Reply:
I thought about this approach, but looks like there is no fairly simple and straightforward ways to do this with with std::set. The problem is that there are may be several kernels with the same priority, so, we need another properties for comparation and sorting except this:
```
            [](const PriorityPair& firstImpl, const PriorityPair& secondImpl) {
                return firstImpl.first < secondImpl.first;
            });
```
Otherwise only one kernel with each priority value will be inserted in the set and all other kernels will be ignored.

Reply:
What about std::multiset then?

Reply:
Sure, thanks